tools/python_api/src_cpp/py_connection.cpp

```diff
@@ -175,10 +176,10 @@ std::unordered_map<std::string, std::unique_ptr<Value>> transformPythonParameter
 }
 
 Value transformPythonValue(py::handle val) {
-    auto datetime_mod = py::module::import("datetime");
-    auto datetime_datetime = datetime_mod.attr("datetime");
-    auto time_delta = datetime_mod.attr("timedelta");
-    auto datetime_date = datetime_mod.attr("date");
+    py::gil_scoped_acquire acquire;
```

Comment:
I don't think you have to hold a gil there.
You only have to hold the gil when the cache is not loaded?

Reply:
I suppose this means I don't have to hold the gil for the last two imports?

Reply:
Duckdb seems to hold the GIL regardless of whether or not it was cached. The acquire statement is just found somewhere else, usually a few function calls ahead of the current function